jlab-sensing/soil-power-sensor-firmware
# 45 Fixing current adc

---
name: Stephen Taylor
about: Implementing Current Readings
title: Junior Specialist 
labels: ''
assignees: 
reviewers: 'jmadden173'
---

**Name/Affiliation/Title**
Stephen Taylor/UCSC/JS

**Purpose of the PR**
Fixed the timing issues with HAL that prevented current readings. Cleaned up libraries to match style guide.

**Development Environment**
Windows, SPS 2.2

**Test Procedure**
Run the unit test, this should test whether the ADC can even function. Then hook the SPS current and voltage ports and run the 'example_adc' environment to check for reasonable values.

**Additional Context**
N/A
